components/camera/camera_test.go

```diff
@@ -250,3 +251,172 @@ func TestCameraWithProjector(t *testing.T) {
+}
+
+func (c *testCamera) Image(ctx context.Context, mimeType string, extra map[string]interface{}) ([]byte, camera.ImageMetadata, error) {
+	return c.imageFunc(ctx, mimeType, extra)
+}
+
+func (c *testCamera) Images(ctx context.Context) ([]camera.NamedImage, resource.ResponseMetadata, error) {
+	if c.imagesFunc != nil {
+		return c.imagesFunc(ctx)
+	}
+	return nil, resource.ResponseMetadata{}, errors.New("not implemented")
+}
+
+func (c *testCamera) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, error) {
+	return nil, errors.New("not implemented")
+}
+
+func (c *testCamera) Properties(ctx context.Context) (camera.Properties, error) {
+	return camera.Properties{}, nil
+}
+
+func (c *testCamera) Close(ctx context.Context) error {
+	return nil
+}
+
+func (c *testCamera) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
+	return nil
+}
+
+// verifyImageEquality compares two images and verifies they are identical.
+func verifyImageEquality(t *testing.T, img1, img2 image.Image) {
+	t.Helper()
+	diff, _, err := rimage.CompareImages(img1, img2)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, diff, test.ShouldEqual, 0)
+}
+
+// verifyDecodedImage verifies that decoded image bytes match the original image.
+func verifyDecodedImage(t *testing.T, imgBytes []byte, mimeType string, originalImg image.Image) {
+	t.Helper()
+	test.That(t, len(imgBytes), test.ShouldBeGreaterThan, 0)
+
+	// For JPEG, compare the raw bytes after encoding the original image
+	if mimeType == rutils.MimeTypeJPEG {
+		expectedBytes, err := rimage.EncodeImage(context.Background(), originalImg, mimeType)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, imgBytes, test.ShouldResemble, expectedBytes)
+		return
+	}
+
+	// For other formats, compare the decoded images
+	decodedImg, err := rimage.DecodeImage(context.Background(), imgBytes, mimeType)
+	test.That(t, err, test.ShouldBeNil)
+	verifyImageEquality(t, decodedImg, originalImg)
+}
+
+func TestGetImagesFromGetImage(t *testing.T) {
+	testImg := image.NewRGBA(image.Rect(0, 0, 100, 100))
+
+	testCam := &testCamera{
+		Named: camera.Named("test_cam").AsNamed(),
+		imageFunc: func(ctx context.Context, mimeType string, extra map[string]interface{}) ([]byte, camera.ImageMetadata, error) {
+			imgBytes, err := rimage.EncodeImage(ctx, testImg, mimeType)
+			if err != nil {
+				return nil, camera.ImageMetadata{}, err
+			}
+			return imgBytes, camera.ImageMetadata{MimeType: mimeType}, nil
+		},
+	}
+
+	t.Run("PNG mime type", func(t *testing.T) {
+		images, metadata, err := camera.GetImagesFromGetImage(context.Background(), rutils.MimeTypePNG, testCam)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, len(images), test.ShouldEqual, 1)
+		test.That(t, images[0].SourceName, test.ShouldEqual, "")
+		verifyImageEquality(t, images[0].Image, testImg)
+		test.That(t, metadata.CapturedAt.IsZero(), test.ShouldBeFalse)
```

Comment:
it should be between the start of the test and the end of the test. not just non-zero

Reply:
Changed